content/docs/rfcs/62/README.md

```diff
@@ -0,0 +1,1077 @@
+
+### ContactUpdate
+
+`ContactUpdate` is a message exchange to notify peers that either the user has been added as a contact, or that information about the sending user have changed.
+
+```protobuf
+message ContactUpdate {
+  uint64 clock = 1;
+  string ens_name = 2;
+  string profile_image = 3;
+  string display_name = 4;
+  uint64 contact_request_clock = 5;
+  ContactRequestPropagatedState contact_request_propagated_state = 6;
+  string public_key = 7;
+}
+
+message ContactRequestPropagatedState {
+  uint64 local_clock = 1;
+  uint64 local_state = 2;
+  uint64 remote_clock = 3;
+  uint64 remote_state = 4;
+}
+```
+
+#### Payload
+
+| Field | Name | Type | Description |
+| ----- | ---- | ---- | ---- |
+| 1 | clock | `uint64` | The clock of the chat with the user |
+| 2 | ens_name | `string` | The ENS name if set |
+| 3 | profile_image | `string` | The base64 encoded profile picture of the user |
+| 4 | display_name | `string` | The display name of the user |
+| 5 | contact_request_clock | `uint64` | The clock at which a contact request was sent |
+| 6 | contact_request_propagated_state | `ContactRequestPropagatedState` | Additional propagation clock values |
+| 7 | public_key | `string` | The public key of the user |
+
+#### Contact update
+
+A client SHOULD send a `ContactUpdate` to all the contacts each time:
+
+- The ens_name has changed
+- A user edits the profile image
+
+A client SHOULD also periodically send a `ContactUpdate` to all the contacts, the interval is up to the client, the Status official client sends these updates every 48 hours.
+
+### EmojiReaction
+
+`EmojiReaction`s represents a user's "reaction" to a specific chat message. 
+For more information about the concept of emoji reactions see [Facebook Reactions](https://en.wikipedia.org/wiki/Facebook_like_button#Use_on_Facebook).
+
+This specification RECOMMENDS that the UI/UX implementation of sending `EmojiReactions` requires only a single click operation, as users have an expectation that emoji reactions are effortless and simple to perform.  
+
+```protobuf
+message EmojiReaction {
+  // clock Lamport timestamp of the chat message
+  uint64 clock = 1;
+
+  // chat_id the ID of the chat the message belongs to, for query efficiency the chat_id is stored in the db even though the
+  // target message also stores the chat_id
+  string chat_id = 2;
+
+  // message_id the ID of the target message that the user wishes to react to
+  string message_id = 3;
+
+  // message_type is (somewhat confusingly) the ID of the type of chat the message belongs to
+  MessageType message_type = 4;
+
+  // type the ID of the emoji the user wishes to react with
+  Type type = 5;
+
+  enum Type {
+    UNKNOWN_EMOJI_REACTION_TYPE = 0;
+    LOVE = 1;
+    THUMBS_UP = 2;
+    THUMBS_DOWN = 3;
+    LAUGH = 4;
+    SAD = 5;
+    ANGRY = 6;
+  }
+
+ // whether this is a retraction of a previously sent emoji
+  bool retracted = 6;
```

Comment:
should all the other data be sent even though it is just a retraction? what is  the intended behaviour when retracted = true, but the `type` is different than the `type` they are retracting?